src/gsy_e/models/strategy/update_frequency.py

```diff
@@ -225,24 +243,17 @@ def set_parameters(self, *, initial_rate: float = None, final_rate: float = None
                        update_interval: int = None) -> None:
         """Update the parameters of the class without the need to destroy and recreate
         the object."""
-        should_update = False
         if initial_rate is not None:
             self.initial_rate_input = initial_rate
-            should_update = True
         if final_rate is not None:
             self.final_rate_input = final_rate
-            should_update = True
         if energy_rate_change_per_update is not None:
             self.energy_rate_change_per_update_input = energy_rate_change_per_update
-            should_update = True
         if fit_to_limit is not None:
             self.fit_to_limit = fit_to_limit
-            should_update = True
         if update_interval is not None:
             self.update_interval = update_interval
-            should_update = True
-        if should_update:
-            self._read_or_rotate_rate_profiles()
+        self._read_or_rotate_rate_profiles()
```

Comment:
I was first afraid that this would increase the computational effort and the number of queries to the profile DB. But there is an if check and the population is skipped if the current time_slot is not part of the profile. If this is only a corner case and this change does not mean that the profiles are still only populated every simulation-week (and not every market slot), all good.

Reply:
This change is not really that different from the former implementation. The only difference between the 2 is that the old implementation would not call the `_read_or_rotate_profiles` method if `set_parameters` was called without arguments (because this is the only condition where should_update remains False). As for the impact to the queries to the DB, it is not different from the previous implementation in the sense that it performs queries only when the current time_slot is not part of the profile, as you mentioned. So I think that we are good here